flink-java/src/main/java/org/apache/flink/api/java/operators/SortedGrouping.java

```diff
@@ -156,6 +156,23 @@ public SortedGrouping(DataSet<T> set, Keys<T> keys, String field, Order order) {
 		return new GroupReduceOperator<T, R>(this, resultType, dataSet.clean(reducer), Utils.getCallLocationName() );
 	}
 
+	/**
+	 * Applies a partial GroupReduce transformation on a grouped and sorted {@link DataSet}.
+	 *
+	 * In contrast to the reduceGroup transformation, the GroupReduce function is only called on each partition. Thus,
+	 * partial solutions are likely to occur.
+	 * @param reducer The ReduceFunction that is applied on the DataSet.
+	 * @return A GroupReducePartial operator which represents the partially reduced DataSet.
+	 */
+	public <R> GroupReducePartialOperator<T, R> reduceGroupPartially(GroupReduceFunction<T, R> reducer) {
+		if (reducer == null) {
+			throw new NullPointerException("GroupReduce function must not be null.");
+		}
+		TypeInformation<R> resultType = TypeExtractor.getGroupReduceReturnTypes(reducer, this.getDataSet().getType());
+
+		return new GroupReducePartialOperator<T, R>(this, resultType, dataSet.clean(reducer), Utils.getCallLocationName());
+	}
+
```

Comment:
Why do we have this here. Partial GroupReduce doesn't make sense on a grouping. Thats's what a regular GroupReduce is. Or am I missing something?